### PR TITLE
Removing the dependency to psutil

### PR DESCRIPTION
**Pull request by gthvn1**
Just reports an error if we can close the qemu_dp process.

### CODE REVIEW

**Review by stormi — COMMENTED** [review]
Would it be worth keeping the old code as a comment to ressurect it later if needed?

Or a simple comment above the `raise` call.

**Comment by gthvn1** [comment]
If we want to ressurect it's in the git log already. But yes if you want I can just comment the code.

**Comment by stormi** [comment]
I'm just asking the question. You and Yann decide :)

**Comment by gthvn1** [comment]
let's wait for yann comment then :)

File: README.md
Comment (ydirson) [inline README.md]: qemu-dp is in the same bucket, but I wouldn't spend too much time on the readme for a codebase that we're highly likely to scrap anyway - not mentioning that if we happen to resurrect qemu-dp on this codebase we risk to forget adding them back

**Review by ydirson — COMMENTED** [review]
The whole qemu-dp support is broken, I would either comment out the whole qemudisk stuff or remove the whole files, rather than just removing that small part of an otherwise-obsolete code

**Comment by gthvn1** [comment]
Removing all code implies to modify the datapath. I think that just for removing the dependency (that is the goal here) it is not needed to also modified the datapath.py

**Comment by ydirson** [comment]
> Removing all code implies to modify the datapath.

That may indeed make a point for #13, after this all we need to be able not to ship `libs/qemudisk.py` (sole user of `psutil`) is to comment out the qcow2 support in `libcow.imageformat`, that would be much more of a surgical strike.

**Comment by stormi** [comment]
What's next on this topic?

**Comment by gthvn1** [comment]
This review will be abandoned probably because @ydirson  is doing restructuring #13 that will allow to not install qemu-dp part and so it will remove the dependency to psutil until we fix qemu-dp. @Wescoeur  and I already reviewed the restructuring PR

**Comment by ydirson** [comment]
It could just be redone on top of #13 without opening a separate PR

**Comment by gthvn1** [comment]
Ok I will do that :+1:

**Comment by gthvn1** [comment]
> The whole qemu-dp support is broken, I would either comment out the whole qemudisk stuff or remove the whole files, rather than just removing that small part of an otherwise-obsolete code

In fact I thought that removing the wholes files was part of your restructuring so that's why I think to abandon this PR. I will wait until you merge #13 and will see.

**Comment by gthvn1** [comment]
rebased on last master branch

**Comment by stormi** [comment]
So, if later qemu-dp gets used again, will psutil come back along with its open CVE (CC @bleader)

**Review by stormi — APPROVED** [review]

**Comment by gthvn1** [comment]
@ydirson are you agree with this or do you still want that we don't ship the file at all. I agree to not ship the file at all but it should be done when building the RPM no? So IIUC if we want to follow this path I need to comment all stuff related to qcow2 in libcow/imageformat.py and then modify the RPM to not ship the file :thinking: ?

**Comment by ydirson** [comment]
Changing the implementation of `qcow2util` would leave the possibility to call that broken implementation.
Disabling `IMAGE_QCOW2` from `libcow.imageformat` would both be less intrusive and avoid this completely, allowing not to ship the whole lot of `q*` modules

**Comment by gthvn1** [comment]
So if I understand correctly instead of commenting the code of _kill_qemu I should instead remove the support of IMAGE_QCOW2. Then once merged we will be able to modify the SRPM to not ship the `libs/qemudisk.py` and others `q*` modules right. I will do that.

**Review by ydirson — APPROVED** [review]
lgtm

File: xapi/storage/libs/libcow/coalesce.py
Comment (stormi) [inline xapi/storage/libs/libcow/coalesce.py]:
```suggestion
                # reenable it...
```

File: xapi/storage/libs/libcow/coalesce.py
Comment (stormi) [inline xapi/storage/libs/libcow/coalesce.py]:
```suggestion
                # Support of QCOW2 has been disabled, comment the code until we
```

**Review by stormi — COMMENTED** [review]
Is the PR still a draft PR on purpose?

**Comment by gthvn1** [comment]
> Is the PR still a draft PR on purpose?

Yes it is on purpose because I made the modifications to be sure that I understand what Yann wanted but I didn't test it yet (I mean install without `q*` files and check that it is working as expected). I will test that today I think.

**Comment by gthvn1** [comment]
tested and ready to merge

**Pull request merged**
